test-src/api_test.js

```diff
@@ -0,0 +1,174 @@
+let workerType = require('../lib/worker-type');
+let workerState = require('../lib/worker-state');
+var assume = require('assume');
+var helper = require('./helper');
+var slugid = require('slugid');
+var assume = require('assume');
+var taskcluster = require('taskcluster-client');
+var _ = require('lodash');
+
+describe('worker-type API methods', () => {
+  let WorkerType, WorkerState, wt;
+
+  let cleanTables = async () => {
+    // remove all rows from either table
+    await WorkerType.scan({} , {
+      handler: async (item) => { await item.remove();  },
+    });
+    await WorkerState.scan({} , {
+      handler: async (item) => { await item.remove();  },
+    });
+  };
+
+  before(async () => {
+    WorkerState = workerState.setup({
+      table: helper.cfg.get('provisioner:workerStateTableName'),
+      credentials: helper.cfg.get('azure'),
+    });
+
+    let keyPrefix = helper.cfg.get('provisioner:awsKeyPrefix');
+    let pubKey = helper.cfg.get('provisioner:awsInstancePubkey');
+    let provisionerId = helper.cfg.get('provisioner:id');
+    let provisionerBaseUrl = helper.cfg.get('server:publicUrl') + '/v1';
+    WorkerType = workerType.setup({
+      table: helper.cfg.get('provisioner:workerTypeTableName'),
+      credentials: helper.cfg.get('azure'),
+      context: {keyPrefix, provisionerId, provisionerBaseUrl, pubKey},
+    });
+
+    await WorkerState.ensureTable();
+    await WorkerType.ensureTable();
+    await cleanTables();
+
+    wt = slugid.nice();
+  });
+
+  afterEach(async () => {
+    await cleanTables();
+  });
+
+  let addTestWorkerType = async () => {
+    await WorkerType.create(wt, {
```

Comment:
We have a module for creating mock worker types.  Please use that instead so that when we change the schema or entity, we don't need to make changes all over the place.  `manageworkertype_test.js` has example implementations.